File: examples/greeter/cli/ruby/sidecar_client.rb
require 'net/http'
require 'json'

# Sidecar Client example in ruby
#
# This speaks to the service go.micro.srv.greeter
# via the sidecar application HTTP interface

uri = URI("http://localhost:8081/rpc")

req = {
  "service" => "ruby.micro.srv.greeter",
  "method"  => "Say.Hello",
  "request" => {"name" => "John"}
}

# do request
http = Net::HTTP.new(uri.host, uri.port)
request = Net::HTTP::Post.new(uri.request_uri)
request.content_type = 'application/json'
request.body = req.to_json

puts JSON.parse(http.request(request).body)

File: examples/greeter/cli/python/sidecar_client.py
import requests
import json


def main():
    url = "http://localhost:8081/rpc"
    headers = {'content-type': 'application/json'}

    # Example echo method
    payload = {
	"service": "go.micro.srv.greeter",
        "method": "Say.Hello",
        "request": {"name": "John"},
    }
    response = requests.post(
        url, data=json.dumps(payload), headers=headers).json()

    print response

if __name__ == "__main__":
    main()
